clientconn.go

```diff
@@ -261,6 +192,38 @@ func (s ConnectivityState) String() string {
 
 // ClientConn represents a client connection to an RPC service.
 type ClientConn struct {
+	picker Picker
+}
+
+// State returns the connectivity state of the Conn used for next upcoming RPC.
+// This is EXPERIMENTAL API.
+func (cc *ClientConn) State() ConnectivityState {
+	c := cc.picker.Peek()
+	if c == nil {
```

Comment:
Go APIs should not return sentinel values like nil.  Return (conn, bool) or (conn, error) instead.